modules/externalipam/phpipam/phpipam_client.rb

```diff
@@ -0,0 +1,168 @@
+require 'yaml'
+require 'json'
+require 'net/http'
+require 'uri'
+require 'sinatra'
+require 'externalipam/externalipam'
+require 'externalipam/ipam_helper'
+require 'externalipam/ipam_validator'
+require 'externalipam/api_resource'
+require 'externalipam/ip_cache'
+
+module Proxy::Phpipam
+  # Implementation class for External IPAM provider phpIPAM
+  class PhpipamClient
+    include Proxy::Log
+    include Proxy::Ipam::IpamHelper
+    include Proxy::Ipam::IpamValidator
+
+    def initialize(conf)
+      @conf = conf
+      @api_base = "#{@conf[:url]}/api/#{@conf[:user]}/"
+      @token = authenticate
+      @api_resource = Proxy::Ipam::ApiResource.new(api_base: @api_base, token: @token, auth_header: 'Token')
+      @ip_cache = Proxy::Ipam::IpCache.instance
+      @ip_cache.provider_name('phpipam')
+    end
+
+    def get_ipam_subnet(cidr, group_name = nil)
+      if group_name.nil? || group_name.empty?
+        get_ipam_subnet_by_cidr(cidr)
+      else
+        group = get_ipam_group(group_name)
+        get_ipam_subnet_by_group(cidr, group[:id])
+      end
+    end
+
+    def get_ipam_subnet_by_group(cidr, group_id)
+      subnets = get_ipam_subnets(group_id)
+      return nil if subnets.nil?
+      subnet_id = nil
+
+      subnets.each do |subnet|
+        subnet_cidr = "#{subnet[:subnet]}/#{subnet[:mask]}"
+        subnet_id = subnet[:id] if subnet_cidr == cidr
+      end
+
+      return nil if subnet_id.nil?
+      response = @api_resource.get("subnets/#{subnet_id}/")
```

Comment:
I think this should also work and provide the same guarantees:
```suggestion
      subnet = subnets&.find { |subnet| cidr == "#{subnet[:subnet]}/#{subnet[:mask]}" }
      return nil if subnet.nil? || subnet[:id].nil?

      response = @api_resource.get("subnets/#{subnet[:id]}/")
```